crates/nu-command/src/filesystem/uutil-notes.md

```diff
@@ -0,0 +1,120 @@
+# Using uu_cratename in nushell
+
+## Argument Passing
+
+We need to get arguments form nushell and pass arguments to uumain. I seriously doubt that the nushell argument structures will be 100% compatible with clap, which is what is used in uutils, so there could be some rough edges.
+
+This is a prototype for copy in PR #9463
+```rust
+fn run(
+    &self,
+    engine_state: &EngineState,
+    stack: &mut Stack,
+    call: &Call,
+    _input: PipelineData,
+) -> Result<PipelineData, ShellError> {
+    // Get the App
+    // let mut app = uu_cp::uu_app();
+    // app.print_help()?;
+
+    // Create uucore::Args somehow from nushell args
+    let s1 = "cp".to_string();
+    let s2 = "-h".to_string();
+    let args = vec![OsString::from(s1), OsString::from(s2)];
+    // Pass uucore::Args to app.uumain
+    uu_cp::uumain(&mut args.into_iter());
+    Ok(PipelineData::empty())
```

Comment:
Same question for output. Here we're dropping whatever they give us back, but how do we convert the uumain's stdout to something nushell can use? or, more importantly, can we convert it to structured data?

Reply:
My first thought was to start with commands like `cp` that don't really have structured output. The problem though is with error messages. We can catch parameter errors via our normal command argument parsing, but once we hand things off to `blah::uumain()` it does what it will do. This is the part we need to work with the uutils team on, having a way to hook the output.

You can see what `uu_cp::uumain()` is doing here https://github.com/uutils/coreutils/blob/17830ad6f84583a3d552b360dece0c74d7efc2e3/src/uu/cp/src/cp.rs#L651-L696

I think it's a good idea to somehow integrate with clap. It looks like 90% of what's going on in that cp.rs file I linked above is all clap/argument handling. So, we'd want to use that to our benefit somehow.